### DISCUSSION TITLE
Conditions using checkbox

### DISCUSSION THREAD

**Discussion by starmomou** [Budibase/budibase#3513] [Help] [answered]
Hi, I'm trying to build a CRUD app based on Budibase template layout.

**Step 1**: 
This is the table where users can update their information throught `Edit`. Assuming that a user want to update the `Contact No`
<img width="1379" alt="螢幕截圖 2021-11-24 上午11 48 28" src="https://user-images.githubusercontent.com/73826858/143172721-9303ede4-93a4-48e8-bf93-a0dd9e92f8c8.png">

**Step 2**:
When the checkbox `Contact No` is ticked, the user is able to see the current `Contact No`
<img width="712" alt="螢幕截圖 2021-11-24 上午11 50 23" src="https://user-images.githubusercontent.com/73826858/143173294-d05f0bb9-acce-48bd-87cb-dd1bcc7c3978.png">

**Step 3**: 
Enter a new `Contact No`
<img width="710" alt="螢幕截圖 2021-11-24 上午11 52 25" src="https://user-images.githubusercontent.com/73826858/143173425-e9158af6-ead8-403a-a3ca-a62d2961fbae.png">

**Step 4**: 
Untick the `Contact No`checkbox
<img width="682" alt="螢幕截圖 2021-11-24 上午11 54 09" src="https://user-images.githubusercontent.com/73826858/143173917-f559b744-88d2-41df-998b-8f809f855198.png">

**Step 5**:
Click `Save` and we can see the `Contact No` has been updated.
<img width="1381" alt="螢幕截圖 2021-11-24 上午11 56 11" src="https://user-images.githubusercontent.com/73826858/143174023-aac699b4-f1f9-43b4-80f0-5db32f4481bb.png">

**Question: How can I keep the the `Contact No` unchanged?**

It would be the best if I can use `Update Setting` with bindings. Thank you

**Comment by mjashanks** [comment, marked as answer]
Hi @starmomou 

In your `Save Row` action, you need to override the `Contact No` field, so that
- If the checkbox is checked, use the value from the input
- if not checked, use the original value

**Override the Contact No. column, using a Javascript function....**

![image](https://user-images.githubusercontent.com/3524181/143249859-8cbe9e10-cadb-46ae-9db9-05cf53fbf205.png)

**My function....**

```javascript
if $("Form.Fields.chngeContactNumber") {
  return $("Form.Fields.Number");
} else {
  return $("Repeater.Person.Number");
}
```

I.e. the `Repeater.Person.Number` is the original value, `Form.Fields.Number` is the value from the input.

**Reply by starmomou** [reply]
Awesome! Much appreciated! Now I know how the parameters work with JavaScript function on Budibase. 
There are so many things can be done 💯

**Reply by melohagan** [reply]
Just a small update to this one. This approach will still work absolutely fine, but there is now another option:

Add the [Update Field Value](https://docs.budibase.com/docs/form-actions#update-field-value) action to the checkbox [On Change](https://docs.budibase.com/docs/form-actions#on-change) event.

You can use the [Continue If](https://docs.budibase.com/docs/chaining-actions#continue-if--stop-if) action to check if the checkbox is being set to blank (using the `{{ Field Value }}` binding, and then set the *Type* of the Update Field Value action to 'Reset to default value'.
